worlds/tunic/__init__.py

```diff
@@ -412,12 +429,14 @@ def fill_slot_data(self) -> Dict[str, Any]:
             "disable_local_spoiler": int(self.settings.disable_local_spoiler or self.multiworld.is_race),
         }
 
+        self.item_link_potential_locs = {}
```

Comment:
I'm thinking that something like this could be an alternative:
Build a dictionary of lists of locations that contain an item belonging to any group that the TUNIC world belongs to, and key those lists of locations by the name of the item at each location. This iterates once through the locations in multiworld for each TUNIC world, but only when the TUNIC world actually belongs to a group.

I think groups are currently only used for item links. Though, checking if a group actually uses item links could be done with `if "item_pool" in self.multiworld.groups[group_id]` though this is somewhat of an implementation detail. Maybe `if self.multiworld.groups[group_id].get("item_pool", ())` might be better.
```py
        groups = self.multiworld.get_player_groups(self.player)
        all_group_item_locations_by_item_name: Dict[str, List[Location]] = {}
        if groups:
            # Some items may be item linked. The name of the item link location itself is not useful to us, so we will
            # try to find the real locations of each item linked item.
            for location in self.multiworld.get_locations():
                item = location.item
                if item is not None and item.player in groups:
                    all_group_item_locations_by_item_name.setdefault(item.name, []).append(location)
```

and then `get_real_location()` would need to have `all_group_item_locations_by_item_name` passed into it as an argument or `all_group_item_locations_by_item_name` would have to be stored on the TUNIC world instance so that `get_real_location()` can retrieve it. Example of passing in the dictionary as an argument:

```py
    def get_real_location(self, location: Location, group_item_locations: Dict[str, List[Location]]) -> Location:
        # if it's not in a group, it's not in an item link
        if location.player not in self.multiworld.groups or not location.item:
            return location

        try:
            return group_item_locations[location.item.name].pop()
        except (KeyError, IndexError):
            warning(f"TUNIC: Failed to parse item location for in-game hints for {self.player_name}. "
                    f"Using a potentially incorrect location name instead.")
            return location
```

Reply:
Wouldn't this just find the locations in the item link slots, not the locations that a player would need to check to send an item to the item link?

Reply:
It's scanning the multiworld for all locations that contain an item belonging to one of the groups the player is in. An item belonging to a group is an 'item link unlocker' item added to the item pool by item links and filled during progression/remaining fill. A player checking a location containing a group item sends that item to every player in the group.
```py
for location in self.multiworld.get_locations():
    item = location.item
    # Check that the item at the location belongs to one of the groups the player is in.
    # When this location is checked, all players in the group receive a copy of the item.
    if item is not None and item.player in groups:
```